### ISSUE TITLE
xAxis update

### ISSUE THREAD

**Issue by sebastianbochan** [highcharts/highcharts#21650]
#### Expected behaviour
Switching between types of xAxis should render ticks according to the type of xAxis.

#### Actual behaviour
1. Demo
The switch from categories to linear does not reset categories

2. Demo
The update from datetime to categories does not set categories properly (timestamps still remain and are converted to strings) 

#### Live demo with steps to reproduce
- https://jsfiddle.net/90o5arkp/3/
- https://jsfiddle.net/90o5arkp/4/

#### Product version
11.4.6

#### Affected browser(s)
All

#### Addtional info
It happened in the past https://github.com/highcharts/highcharts/issues/12534

**Comment by KacperMadej** [comment]
@sebastianbochan 
I have doubts about the 1st demo. The options for xAxis are set directly so they should be changed directly - https://jsfiddle.net/BlackLabel/480msx3a/ - then it works correctly as far as I can see.

In the 2nd case the xAxis options are set indirectly, so they should update when the series data is changed. This feature is missing so it looks like a bug.

**Comment by TorsteinHonsi** [comment]
It's not necessarily a bug. If we take a look at Kacper's second demo: https://jsfiddle.net/xv4pj1cf/

What happens is that we are updating from two data point to another two data points, and Highcharts assumes we want to keep the points (with animation and all), so it matches them. Since there are x values defined for the first set of points, but not the second, HC keeps the x values of the first set and applies that on the category axis. Which is kind of how we intended point matching to work, except its result is a bit surprising in this case.

The workaround is simply to define the x values that we do want: https://jsfiddle.net/highcharts/80waLp3v/ . This causes the series logic to create new points because they can't be matched directly. If we want animation, we can add id's: https://jsfiddle.net/highcharts/bn5ythgs/

 : https://jsfiddle.net/highcharts/80waLp3v/

**Comment by KacperMadej** [comment]
- Live steps demo 1: https://jsfiddle.net/90o5arkp/3/
`type: 'category',` is set on `xAxis` directly so the categories are forced (via `forceCategory` that detects precisely this situation) - `data` updates won't change `xAxis`' `type` in this case by design. If this changes then it will be a breaking change. A new option could be added (e.g. `ignoreAxisType?: boolean = false`) to handle this. Could you check if this is actually a real life case that needs fixing?

- Better demos 1: https://jsfiddle.net/Lc4ty9u5/6/
The same as above. 

- Better demos 2: https://jsfiddle.net/xv4pj1cf/
No data module. Points are updated but `x` values are not changed, so they are staying. `Series.setData` can work both ways via `keepPoints` argument but this is not exposed on the `Chart.update` level. Easier fix is to set and/or update `x` values like Torstein explained in his comment above. However, if I update `keys` and series data in the same update it doesn't work as expected, but this looks like a different problem.

-  Live steps demo 1: https://jsfiddle.net/90o5arkp/4/
Some options are set when deduced for xAxis by the data module but not reverted on change and other options are set in core (e.g. pointStart) that can influence how the chart behaves. Those can be worked around when chart config is set directly but when all is deduced from data by the data module it would require extra code in `processed` callback which is suboptimal and in a lot of cases looks very buggy. I have probably fixed some/all and will test with Sebastian to get the full picture.

tl;dr
There are 3 problems.
1st is a data module bug that's in progress of fixing.
2nd is undecided being a feature if used correctly.
3rd is a different bug that I will open a new ticket for after better investigation.
